hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1317 @@
+        std::string config_system_property ;
+        if(!env){
+            config_system_property = "";
+        } else{
+            config_system_property = std::string(env);
+        }
+        if (config_system_property.empty()) {
+            std::cout << "FINEST: "
+                      << "Could not find " + property_key + " environment variable" << std::endl;
+            return false;
+        }
+
+        if (!declarative_config_util::is_accepted_suffix_configured(
+              config_system_property, accepted_suffixes)) {
+            if (fail_on_unaccepted_suffix) {
+                declarative_config_util::
+                  throw_unaccepted_suffix_in_system_property(
+                    property_key, config_system_property, accepted_suffixes);
+            } else {
+                return false;
+            }
+        }
+
+        std::cout << "INFO: "
+                  << "Loading configuration " + config_system_property +
+                       " from environment variable "
+                  << property_key << std::endl;
+        load_system_property_file_resource(config_system_property);
+        return true;
+    } catch (hazelcast::client::exception::hazelcast_& e) {
+        throw;
+    } catch (hazelcast::client::exception::iexception& e) {
+        throw hazelcast::client::exception::hazelcast_(
+          "abstract_config_locator::load_from_system_property",
+          e.get_message(),
+          "",
+          std::current_exception());
+    }
+}
+bool
+xml_client_config_locator::locate_in_work_directory()
+{
+    return load_from_working_directory("hazelcast-client.xml");
+}
+bool
+xml_client_config_locator::locate_from_system_property()
+{
+    return load_from_system_property(
+      "hazelcast.client.config",
+      {"xml"});
+}
+
+bool
+abstract_config_builder::should_validate_the_schema()
+{
+    return false; //validation hasn't been implemented yet
+}
+
+std::string
+abstract_xml_config_helper::get_release_version()
+{
+    return hazelcast::client::version().to_string();
+}
+void
+abstract_xml_config_helper::schema_validation(const boost::property_tree::ptree& doc)
+{
+    // validation hasn't implemented yet.
+}
+
+std::string
+abstract_xml_config_builder::get_attribute(boost::property_tree::ptree node,
+                                           const std::string& att_name)
+{
+    return node.get_child("<xmlattr>." + att_name).data();
+}
+
+void
+abstract_xml_config_builder::process(boost::property_tree::ptree* root)
+{
+    replace_imports(root);
+    replace_variables(root);
+}
+
+void
+abstract_xml_config_builder::replace_variables(
+  boost::property_tree::ptree* root)
+{
+    bool fail_fast = false;
+    std::vector<property_replacer> replacers;
+    boost::property_tree::ptree node;
+    try {
+        node = root->get_child("config-replacers");
+        try {
+            std::string fail_fast_attr =
+              get_attribute(node, "fail-if-value-missing");
```

Comment:
If 'fail-if-value-missing' attribute is missing, then it doesn't parse the replacers but at Java side, it continues to parse and defaults 'fail-if-value-missing' attribute to 'true'